src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/toolwindow/TestGenieToolWindow.kt

```diff
@@ -226,16 +206,30 @@ class TestGenieToolWindow {
     private fun saveState() {
         val state: TestGenieToolWindowState = TestGenieToolWindowService.getInstance().state!!
 
-        state.searchBudget = searchBudget.value as Int
-        state.localSearchBudgetType = localSearchBudgetType.item
-        state.localSearchBudgetValue = localSearchBudgetValue.value as Int
         state.stoppingCondition = stoppingCondition.item
+        state.searchBudget = searchBudget.value as Int
         state.initializationTimeout = initializationTimeout.value as Int
         state.minimizationTimeout = minimisationTimeout.value as Int
         state.assertionTimeout = assertionTimeout.value as Int
         state.junitCheckTimeout = junitCheckTimeout.value as Int
-        state.population = population.value as Int
         state.populationLimit = populationLimit.item
+        state.population = population.value as Int
+    }
+
+    /**
+     * Adds tooltips to the actual UI elements, not the labels for them.
+     */
+    private fun addTooltipsToUiElements() {
+        stoppingCondition.toolTipText = "What condition should be checked to end the search."
+        searchBudget.toolTipText = "Maximum search duration."
+        initializationTimeout.toolTipText = "Seconds allowed for initializing the search."
+        minimisationTimeout.toolTipText = "Seconds allowed for minimization at the end."
+        assertionTimeout.toolTipText = "Seconds allowed for assertion generation at the end."
+        junitCheckTimeout.toolTipText = "Seconds allowed for checking the generated JUnit files <p/>(e.g., compilation and stability)."
+        populationLimit.toolTipText = "What to use as limit for the population size."
+        population.toolTipText = "Population size of genetic algorithm."
+
+        resetButton.toolTipText = "Reset all parameters to their default values"
```

Comment:
Were these in your general constructor before? You've had tooltips prior to this, no?

Reply:
We did have tooltips before, however now we refactored it to make the tooltips clearer to see.